### ISSUE TITLE
Only the last modified label appears to be drawn in 2.0.dev10

### ISSUE THREAD

**Issue by Square789** [pyglet/pyglet#503]
```py
from string import ascii_uppercase
from random import randint

import pyglet

text = [
    "According",
    "to all known",
    "laws of",
    "aviation",
]

w = pyglet.window.Window()

batch = pyglet.graphics.Batch()

labels = [
    pyglet.text.Label(
        s,
        x = 10,
        y = 100 - i*18,
        font_size = 14,
        batch = batch,
    ) for i, s in enumerate(text)
]

LETTER_MAP = {getattr(pyglet.window.key, x): x for x in ascii_uppercase}

@w.event
def on_draw():
    w.clear()
    batch.draw()

@w.event
def on_key_press(sym, mod):
    try:
        letter = LETTER_MAP[sym]
    except KeyError:
        return
    labels[randint(0, len(text) - 1)].text = letter

pyglet.app.run()
```
In the newest dev10 release, only the label that has most recently been modified/created seems to actually be rendered by a call to `batch.draw()`.

Here's two videos showcasing the problem:

https://user-images.githubusercontent.com/46634729/139608331-4f5c3c1e-2356-4a2e-bc15-f4a3e553c308.mp4

https://user-images.githubusercontent.com/46634729/139608335-b5611d5d-8989-4ce2-93da-b5b21d838748.mp4

**Comment by einarf** [comment]
I'm also seeing issues with text disappearing.

**Comment by benmoran56** [comment]
This is now fixed in the 2.0.dev11 release.
